cmd/seq-coordinator-manager/seq-coordinator-manager.go

```diff
@@ -0,0 +1,308 @@
+			key := seqManager.nonPriorityList[index]
+			seqManager.priorityList = append(seqManager.priorityList, key)
+			seqManager.prioritiesMap[key]++
+
+			index = len(seqManager.priorityList) - 1
+			seqManager.updatePriorityList(ctx, index, target)
+
+			nonPriorityForm.Clear(true)
+			seqManager.populateLists(ctx)
+			pages.SwitchToPage("Menu")
+			if len(seqManager.nonPriorityList) > 0 {
+				app.SetFocus(nonPrioritySeqList)
+			} else {
+				app.SetFocus(prioritySeqList)
+			}
+		})
+		nonPriorityForm.AddButton("Cancel", func() {
+			nonPriorityForm.Clear(true)
+			pages.SwitchToPage("Menu")
+			app.SetFocus(nonPrioritySeqList)
+		})
+		nonPriorityForm.SetFocus(0)
+		app.SetFocus(nonPriorityForm)
+	})
+
+	// UI design
+	flex := tview.NewFlex()
+	priorityHeading := tview.NewTextView().
+		SetTextColor(tcell.ColorYellow).
+		SetText("-----Priority List-----")
+	nonPriorityHeading := tview.NewTextView().
+		SetTextColor(tcell.ColorYellow).
+		SetText("-----Not in priority list but online-----")
+	instructions := tview.NewTextView().
+		SetTextColor(tcell.ColorYellow).
+		SetText("(r) to refresh\n(s) to save all changes\n(c) to switch between lists\n(a) to add sequencer\n(q) to quit\n(tab) to navigate")
+
+	flex.SetDirection(tview.FlexRow).
+		AddItem(priorityHeading, 0, 1, false).
+		AddItem(tview.NewFlex().
+			AddItem(prioritySeqList, 0, 2, true).
+			AddItem(priorityForm, 0, 3, true), 0, 12, true).
+		AddItem(nonPriorityHeading, 0, 1, false).
+		AddItem(tview.NewFlex().
+			AddItem(nonPrioritySeqList, 0, 2, true).
+			AddItem(nonPriorityForm, 0, 3, true), 0, 12, true).
+		AddItem(instructions, 0, 3, false).SetBorder(true)
+
+	flex.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		if event.Rune() == 114 {
+			seqManager.refreshAllLists(ctx)
+			priorityForm.Clear(true)
+			nonPriorityForm.Clear(true)
+			seqManager.populateLists(ctx)
+			pages.SwitchToPage("Menu")
+			app.SetFocus(prioritySeqList)
+		} else if event.Rune() == 115 {
+			seqManager.pushUpdates(ctx)
+			priorityForm.Clear(true)
+			nonPriorityForm.Clear(true)
+			seqManager.populateLists(ctx)
+			pages.SwitchToPage("Menu")
+			app.SetFocus(prioritySeqList)
+		} else if event.Rune() == 97 {
+			addSeqForm.Clear(true)
+			seqManager.addSeqPriorityForm(ctx)
+			pages.SwitchToPage("Add Sequencer")
+		} else if event.Rune() == 99 {
+			if prioritySeqList.HasFocus() {
+				app.SetFocus(nonPrioritySeqList)
+			} else {
+				app.SetFocus(prioritySeqList)
+			}
+		} else if event.Rune() == 113 {
+			app.Stop()
+		}
+		return event
+	})
+
+	pages.AddPage("Menu", flex, true, true)
+	pages.AddPage("Add Sequencer", addSeqForm, true, false)
+
+	if err := app.SetRoot(pages, true).EnableMouse(true).Run(); err != nil {
+		panic(err)
+	}
+}
+
+// updatePriorityList updates the list by changing the position of seq present at `index` to target
+func (sm *manager) updatePriorityList(ctx context.Context, index int, target int) {
+	for i := index - 1; i >= target; i-- {
```

Comment:
I am assuming index and target come from UI right, so there should be no chance of out of bounds err for priorityList right? 

If the above understanding is not right please add a check to avoid that.

Reply:
Yes, out-of-bounds condition shouldn't be possible